tower-buffer/src/future.rs

```diff
@@ -1,26 +1,26 @@
 //! Future types
 
-use crate::{
-    error::{Closed, Error},
-    message,
-};
-use futures::{Async, Future, Poll};
+use crate::{error::Error, message};
+use std::future::Future;
+use std::pin::Pin;
+use std::task::{Context, Poll};
 
 /// Future eventually completed with the response to the original request.
 pub struct ResponseFuture<T> {
     state: ResponseState<T>,
 }
+impl<T> Unpin for ResponseFuture<T> {}
```

Comment:
So, the claim here is that we never move `state`, but I don't think that's true? Specifically, we do `self.state = ` in `poll` below. Why is that okay?

Reply:
That should drop the inner state instead of mem replacing it, right? So in theory that should be safe since we never hand out a &mut T nor a Pin.